app/code/Magento/Contact/Controller/Index/Post.php

```diff
@@ -103,22 +106,25 @@ private function sendEmail($post)
     }
 
     /**
+     * Method to validated params.
+     *
      * @return array
      * @throws \Exception
      */
     private function validatedParams()
     {
         $request = $this->getRequest();
-        if (trim($request->getParam('name')) === '') {
+
+        if (trim($request->getParam('name', '')) === '') {
             throw new LocalizedException(__('Enter the Name and try again.'));
         }
-        if (trim($request->getParam('comment')) === '') {
+        if (trim($request->getParam('comment', '')) === '') {
             throw new LocalizedException(__('Enter the comment and try again.'));
         }
-        if (false === \strpos($request->getParam('email'), '@')) {
+        if (\strpos($request->getParam('email', ''), '@') === false) {
             throw new LocalizedException(__('The email address is invalid. Verify the email address and try again.'));
         }
-        if (trim($request->getParam('hideit')) !== '') {
+        if (trim($request->getParam('hideit', '')) !== '') {
             throw new \Exception();
```

Comment:
Could you please try to change \Exception() to \Magento\Framework\Validation\ValidationException and check that method behavior is not changed ?

Reply:
Hello @xmav, thanks for your review, I fixed code by you recommendation.

The method behavior has not changed, because `trim` from `NULL` return `''`, and the getParam() method is returned value or default value `NULL`.

Thank you.

Reply:
I mean when we change exception class we affect logic in execute() method. 
It looks like we don't have custom exception class to substitute \Exception in this method. Changing to `\Magento\Framework\Validation\ValidationException` will change execute() method flow.
Please rollback to  \Exception. Static test failure will be approved (or skip it if possible)

Reply:
@xmav, I revert change and skipped static failure, please check if every think is well.


Thank you